internal/testrunner/runners/system/tester.go

```diff
@@ -2130,63 +2129,67 @@ func (r *tester) checkTransforms(ctx context.Context, config *testConfig, pkgMan
-					if err != nil {
-						return false, err
-					}
-					foundDeleted := deleted > 0
-					// Wait at least for the number of documents found after running the transform preview
-					foundHits := hits.size() >= len(transformDocs)
-					if foundDeleted {
-						logger.Debugf("Found %d deleted docs in %s index", deleted, destIndexTransform)
-					}
-					if foundHits {
-						logger.Debugf("Found %d hits in %s index", hits.size(), destIndexTransform)
-					}
-					return foundDeleted || foundHits, nil
-				},
-			}
-
-			_, waitErr := r.waitForDocs(ctx, destIndexTransform, waitOpts)
-			if waitErr != nil {
-				return waitErr
+			err := r.validateTransformsWithMappings(ctx, transformId, transform.Name, destIndexTransform, indexTemplateTransform, transformDocs, fieldsValidator)
+			if err != nil {
+				return err
 			}
+		}
+	}
 
-			// As it could happen that there are no hits found , just deleted docs
-			// Here it is used the docs found in the preview API response
-			logger.Warn("Validate mappings found in transform (technical preview)")
-			exceptionFields := listExceptionFields(transformDocs, fieldsValidator)
+	return nil
+}
 
-			mappingsValidator, err := fields.CreateValidatorForMappings(r.esClient,
-				fields.WithMappingValidatorFallbackSchema(fieldsValidator.Schema),
-				fields.WithMappingValidatorIndexTemplate(indexTemplateTransform),
-				fields.WithMappingValidatorDataStream(destIndexTransform),
-				fields.WithMappingValidatorExceptionFields(exceptionFields),
-			)
+func (r *tester) validateTransformsWithMappings(ctx context.Context, transformId, transformName, destIndexTransform, indexTemplateTransform string, transformDocs []common.MapStr, fieldsValidator *fields.Validator) error {
+	// In order to compare the mappings, it is required to wait until the documents has been
+	// ingested in the given transform index
+	// It looks like that not all documents ingested previously in the main data stream are going
+	// to be ingested in the destination index of the transform. That's the reason to disable
+	// the asserts
+	waitOpts := waitForDocsOptions{
+		period: 5 * time.Second,
+		// Add specific timeout to ensure transform processes the documents (depends on "delay" field?)
+		timeout: 10 * time.Minute,
+		stopCondition: func(_ *hits, _ int) (bool, error) {
+			processed, err := r.processedDocsByTransform(ctx, transformId)
 			if err != nil {
-				return fmt.Errorf("creating mappings validator for the transform index failed (index: %s): %w", destIndexTransform, err)
+				return false, err
 			}
-
-			if errs := validateMappings(ctx, mappingsValidator); len(errs) > 0 {
-				return testrunner.ErrTestCaseFailed{
-					Reason:  fmt.Sprintf("one or more errors found in mappings in the transform (index %s)", destIndexTransform),
-					Details: errs.Error(),
-				}
+			if processed > 0 {
+				logger.Debugf("Documents processed by transform %q: %d", transformId, processed)
 			}
-		}
+			return processed >= len(transformDocs), nil
```

Comment:
Looks like we are comparing different things here:
* `len(transformDocs)` is the number of documents resulting from the preview.
* `processed` is the number of processed documents from the source data stream/s.

So, if I understand correctly, `processed` can be bigger than `len(transformDocs)` before all the documents in the data stream are processed.

I would expect this to stop when `processed >= number of docs in the source index at some point to be decided`.

Reply:
Yes, I mean to compare with the number of documents in the source data stream. What I am not so sure is when to check for this number.

> This number of documents in the data stream could be added as first step into validateTransformsWithMappings.

If we are sure that we have enough documents at this point it would be perfect, yes.

Reply:
> > This number of documents in the data stream could be added as first step into validateTransformsWithMappings.
> 
> If we are sure that we have enough documents at this point it would be perfect, yes.

Not totally 100% sure, but currently I cannot think of any other better location. This could vary for each package, since there could be packages that are ingesting documents continuously or they do not have ingested all the required documents in the data stream validation (related to https://github.com/elastic/elastic-package/issues/2378).